.github/workflows/publish-web-client-next.yml

```
@@ -1,68 +1,114 @@
+# the 'web-client' package.json file has been updated.
 
 name: Publish Web Client SDK to NPM on Next
 
 on:
-  push:
+  pull_request:
     branches:
       - next
+    types:
+      - closed
 
 permissions:
   contents: read
 
 jobs:
   publish:
+    if: >
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'patch release')
     runs-on: ubuntu-latest
 
     steps:
       - name: Check out code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          ref: next
 
       - name: Set up Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version: '20'
           registry-url: 'https://registry.npmjs.org/'
 
-      - name: Set up Rust and wasm target
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          profile: minimal
-          override: true
-          target: wasm32-unknown-unknown
-          components: rust-src 
+      - name: Set up Rust and wasm target
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          target: wasm32-unknown-unknown
+          components: rust-src
+
+      - name: Verify version bump against next
+        id: check_version
+        run: |
+          set -euo pipefail
+
+          SHOULD_PUBLISH=true
+
+          BASE_SHA="${{ github.event.pull_request.base.sha }}"
+
+          if [ -z "$BASE_SHA" ]; then
+            echo "Base SHA is unavailable; cannot verify version bump."
+            SHOULD_PUBLISH=false
+          fi
+
+          if [ "$SHOULD_PUBLISH" = "true" ]; then
+            if ! git show "$BASE_SHA:crates/web-client/package.json" > /tmp/base_package.json; then
+              echo "Unable to read crates/web-client/package.json from $BASE_SHA."
+              SHOULD_PUBLISH=false
+            fi
+          fi

```

I think this logic is sound already, but some thoughts I had that could potentially make it easier to read and slightly more performant:
* Change `SHOULD_PUBLISH` to default to `false` and flip after all all the checks pass. This seems safer in case there's any ambiguity or weirdness that happens in a step for whatever reason
* Add a short-circuit step that skips the `package.json` file read + comparison in case it didn't change at all compared to base via something like 

```
if ! git diff --name-only "$BASE_SHA"...HEAD -- crates/web-client/package.json | grep -q .; then
  echo "No changes to crates/web-client/package.json; skipping publish."
  exit 0
fi
```
so that the CI fails faster in case there are no changes
* If you decide to do `SHOULD_PUBLISH=false` by default, you can treat each step as a short circuit condition and call `exit 0` instead of changing the variable and having the code keep going through checking the logic gates

Anyway, I think the code works as is, just sharing an alternate approach if you think it makes sense! 👍